mobile/lib/features/trade/domain/trade_values.dart

```diff
@@ -113,7 +132,13 @@ class TradeValues {
   _recalculateQuantity() {
     Usd? quantity =
         tradeValuesService.calculateQuantity(price: price, margin: margin, leverage: leverage);
-    this.quantity = quantity;
+    this.quantity = quantity ?? Usd.zero();
+  }
+
+  // Note, this only works because it is only called from [`updatePriceAndQuantity`], which is only
+  // called if the max quantity lock is set.
+  _recalculateContracts() {
+    contracts = quantity + openQuantity;
```

Comment:
`openQuantity` seems to be always zero?

Reply:
It is set when the trade bottom sheet tab is initialised.